sentry_sdk/transport.py

```diff
@@ -35,12 +40,18 @@
     from urllib import getproxies  # type: ignore
 
 
-class Transport:
+class Transport(ABC):
     """Baseclass for all transports.
 
     A transport is used to send an event to sentry.
     """
 
+    # TODO: For the items without default implementations, we may wish to mark them as abstract methods,
+    #       or provide a default implementation. We could probably move implementations of some of the
+    #       methods (such as `record_lost_event`) from the HttpTransport class to this class.
+    #       However, marking methods as abstract methods is a breaking change, so we should consider
+    #       whether we would like to make the change for the SDK 2.0 release.
+
```

Comment:
What do you all think?

Reply:
Not sure -- I'd only turn a method into an `abstractmethod` if it absolutely needs to be overridden in the child class for the transport to work at all, which is probably only the case for `capture_envelope`. Otherwise we'd be forcing folks to write methods that they might not need just so that they can use their custom Transport.

We _could_ move the definitions of some of the other methods from `HttpTransport` to `Transport`, but then we're making assumptions about users' custom transports that might not be true. I think the parent `Transport` class should be kept minimal, and at the same time we should allow the user subclassing it to define or leave out as much as they want as long as it still defines at least the `capture_envelope` method.

Reply:
Ok, I think I was spending a bit too much time trying to refactor `record_lost_event`, so let's keep it as is for now. I put the work I did so far in a draft PR (#2660). The changes are backwards compatible, so we can wait until later to make the changes.